spacy/tokens/doc.pyx

```diff
@@ -1484,6 +1484,14 @@ cdef class Doc:
                 token_data["dep"] = token.dep_
                 token_data["head"] = token.head.i
             data["tokens"].append(token_data)
+        
+        data["spans"] = {}
```

Comment:
I'd add a conditional here: `if self.spans` so that `data` doesn't have an empty dict if there is no spans data - that will also keep the behaviour exactly the same for use-cases where there is no `spans` data.